src/cli/analytics.rs

```diff
@@ -77,51 +84,148 @@ impl Command for Analytics {
     }
 }
 
+pub struct AnalyticsStream {
+    span: Span,
+    streams: StreamMap<String, RawJsonRowStreamer>,
+    // This allows us to extend the lifetime of the runtime used to create the streams longer than
+    // run(), else we panic when reading the streams
+    rt: Arc<Runtime>,
+}
+
+impl Iterator for AnalyticsStream {
+    type Item = Value;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some((cluster, result)) = self
+            .rt
+            .clone()
+            .block_on(async { self.streams.next().await })
+        {
+            let bytes = match result {
+                Ok(r) => r,
+                Err(e) => {
+                    return Some(Value::Error {
```

Comment:
Does this mean that the iterator is going to try to continue even after an error?

Reply:
No when an error is returned the iterator stops being read, I altered the code to always return an error then ran: 

```
analytics "FROM range(0,1000000) AS r SELECT repeat("testing", 10) AS x1, repeat("testing", 10) AS x2"
```

And only one error was returned, then the stream stopped, instead of seeing 1000000 errors.